src/main/java/de/nplay/moderationbot/moderation/ModerationService.java

```diff
@@ -0,0 +1,148 @@
+         *
+         * @return a {@link RowMapping} of this record
+         */
+        @MappingProvider("")
+        public static RowMapping<Moderation> map() {
+            return row -> new Moderation(
+                    row.getLong("id"),
+                    row.getLong("user_id"),
+                    ModerationType.valueOf(row.getString("type")),
+                    Optional.ofNullable(row.getBoolean("reverted") ? row.getBoolean("reverted") : null),
+                    Optional.ofNullable(row.getString("reason")),
+                    Optional.ofNullable(row.getInt("paragraph_id") == 0 ? null : RuleService.getRuleParagraph(row.getInt("paragraph_id"))),
+                    Optional.ofNullable(row.getLong("reference_message") == 0 ? null : MessageReferenceService.getMessageReference(row.getLong("reference_message"))),
+                    Optional.ofNullable(row.getTimestamp("revoke_at")),
+                    Optional.ofNullable(row.getLong("duration") == 0 ? null : row.getLong("duration")),
+                    row.getLong("issuer_id"),
+                    row.getDate("created_at")
+            );
+        }
+
+        @Override
+        public String toString() {
+            return "Moderation{" +
+                    "id=" + id +
+                    ", userId=" + userId +
+                    ", type=" + type +
+                    ", reverted=" + reverted +
+                    ", reason=" + reason +
+                    ", paragraph=" + paragraph +
+                    ", referenceMessage=" + referenceMessage +
+                    ", revoke_at=" + revoke_at +
+                    ", duration=" + duration +
+                    ", issuerId=" + issuerId +
+                    ", created_at=" + created_at +
+                    '}';
+        }
+    }
+
+    /**
+     * Retrieves a moderation by its ID.
+     *
+     * @param moderationId The ID of the moderation to retrieve.
+     * @return The {@link Moderation} record.
+     */
+    public static Moderation getModeration(long moderationId) {
+        return Query.query("SELECT * FROM moderations WHERE id = ?")
+                .single(Call.of().bind(moderationId))
+                .mapAs(Moderation.class)
+                .first().orElseThrow();
+    }
+
+    /**
+     * Creates a new moderation record in the database.
+     *
+     * @param userId           The ID of the user being moderated.
+     * @param type             The type of moderation.
+     * @param issuerId         The ID of the user issuing the moderation.
+     * @param reason           The reason for the moderation.
+     * @param paragraph        The rule paragraph associated with the moderation.
+     * @param referenceMessage The reference message associated with the moderation.
+     * @param duration         The duration of the moderation.
+     * @return The {@link InsertionResult} (with keys) of the operation.
+     */
+    public static InsertionResult createModeration(
```

Comment:
Do we really need the InserationResult here? We don't return it on other method either

Reply:
It is needed to get the id of the created report. Otherwise it is not possible to retrieve the created moderation inside the /moderate slash command